### DISCUSSION TITLE
unviersal villager heal discounts

### DISCUSSION THREAD

**Discussion by RhythmicSys** [PurpurMC/Purpur#680] [__Dismissed__] [answered]
Toggle to allow villager discounts from healing infected villagers to be applied to all players instead of just the player who healed the villager.
People think this is how it is already, a lot of the time, and end up spending time infecting/healing villagers for public trade halls so they can have discounts, when it only applies to the initial person

**Comment by Mernst0318** [comment]
I.E. An option that would allow Player A to cure a zombie villager multiple times and receive the discount as well as Player B being able to trade with the same villager at the discounted price without having to cure the villager himself.

In my case, this is very useful as a few friends and myself are building a large villager trading hall on a server with 3-4 people and being able to take turns curing the villagers would save significant time and would more easily allow for us to build a large trading hall instead of each of us having small separate trading halls.

The other benefit to this is due to the option to fix zombie villager trading (fix-curing-zombie-villager-discount-exploit: false) seems to not always work 100% of the time and reset the prices for some players on the server.

**Comment by BillyGalbreath** [comment, marked as answer]
This is impossible to implement. The trade prices are based on _all_ the gossip types of a player, and gossip is stored per player _and_ per gossip type. This would cause all sorts of problems. Like, any one player could just go around punching villagers to make everyone's trade prices suffer for the whole village, etc.
